scripts/maintenance/standard-header-guards.pl

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/perl
+
+## Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+# source-maintenance plugin to standardize the header guards
+
+use strict;
+use warnings;
+
+my $filename = $ARGV[0];
+
+if ($filename !~ /\.h$/i || $filename =~ /^scripts\/boilerplate.h/i) {
+    while (<>) { print; }
+    exit(0);
+}
+
+my @accumulate=();
+my $current_guard = undef;
+
+my $first_ifdef_pos = undef;
+my $last_ifdef_pos = undef;
+while (<>) {
+    if (!defined($first_ifdef_pos) && /^#ifndef\s+(\w+_H)/ ) {
+        $current_guard = $1;
+        $first_ifdef_pos = $#accumulate+1;
+    } elsif (/^#endif/) {
+        $last_ifdef_pos = $#accumulate+1;
+    }
+    push(@accumulate, $_);
+}
+
+die("cannot detect header guard in $filename") unless defined($current_guard);
+die("no #endif in $filename") unless defined($last_ifdef_pos);
+die ("#define $current_guard doesn't immediately follow first #ifndef in $filename")
+    unless ($accumulate[$first_ifdef_pos+1] =~ /^#define\s+$current_guard/);
+for (@accumulate[$last_ifdef_pos+1..$#accumulate]) {
+    die("unexpected content '$_' after last #ifndef in $filename") unless(/^$/);
+}
+
+my $new_guard = $filename;
+$new_guard =~ s/[\/\-\.]/_/g;
+$new_guard = "SQUID_".uc($new_guard);
```

Comment:
If we replace guard lines inside the accumulation array, we would not have to split array printing into several groups with complex start/end points. I do not insist on that approach though.